src/cdk/v2/destinations/reddit/utils.js

```diff
@@ -28,6 +28,59 @@ const batchEvents = (successfulEvents) => {
   const batchedEvents = batchEventChunks(eventChunks);
   return batchedEvents;
 };
+
+const calculateDefaultRevenue = (properties) => {
+  // Check if working with products array
+  if (properties?.products && properties.products.length > 0) {
+    // Check if all product prices are undefined
+    if (properties.products.every((product) => product.price === undefined)) {
+      return null; // Return null if all product prices are undefined
+    }
+    // Proceed with calculation if not all prices are undefined
+    return properties.products.reduce(
+      (acc, product) => acc + (product.price || 0) * (product.quantity || 1),
+      0,
+    );
+  }
+  // For single product scenario, check if price is undefined
+  if (properties.price === undefined) {
+    return null; // Return null if price is undefined
```

Comment:
@shrouti1507 Make use of codecov annotations to get 100% coverage for utils

Reply:
src/cdk/v2/destinations/reddit/utils.test.js file contains all of the edge cases. As discussed we will add component test case for each of them next time.